gensrc/thrift/FrontendService.thrift

```diff
@@ -1338,6 +1338,40 @@ struct TUpdateResourceUsageResponse {
     1: optional Status.TStatus status
 }
 
+struct TResourceSlot {
+    1: optional Types.TUniqueId slot_id
+    2: optional Types.TNetworkAddress request_endpoint
+    3: optional i64 group_id
+    4: optional i32 num_slots
+    5: optional i64 expired_pending_time_ms
+    6: optional i64 expired_allocated_time_ms
+}
+
+struct TRequireSlotRequest {
+    1: optional TResourceSlot slot
+}
+
+struct TRequireSlotResponse {
```

Comment:
why need  an empty struct?

Reply:
Because `requireSlot` is an async request, but we only implement and provide `Frontend.Iface`, not `Frontend.AsyncIface`.

Therefore, we must use two method `reuiqreSlot` and `finishSlotRequirement` to simulate async request.